lib/ruby_lsp/requests/support/common.rb

```diff
@@ -147,6 +147,15 @@ def constant_name(node)
         rescue Prism::ConstantPathNode::DynamicPartsInConstantPathError
           nil
         end
+
+        sig { params(node: Prism::ModuleNode).returns(T.nilable(String)) }
+        def module_constant_name(node)
+          path = node.constant_path
+          case path
+          when Prism::ConstantPathNode, Prism::ConstantReadNode, Prism::ConstantPathTargetNode
+            constant_name(path)
+          end
+        end
```

Comment:
Nitpick: let's rename to `namespace_constant_name` because this is the same story for classes.

```suggestion
        sig { params(node: T.any(Prism::ModuleNode, Prism::ClassNode)).returns(T.nilable(String)) }
        def namespace_constant_name(node)
          path = node.constant_path
          case path
          when Prism::ConstantPathNode, Prism::ConstantReadNode, Prism::ConstantPathTargetNode
            constant_name(path)
          end
        end
```